src/main/java/com/hms_networks/sc/cumulocity/api/CConnectorMqttMgr.java

```diff
@@ -94,6 +94,20 @@ public class CConnectorMqttMgr extends ConstrainedMqttManager {
   /** Stack of {@link CConnectorRetryMessage}s which have been queued for retry. */
   private final Stack pendingRetryMessages = new Stack();
 
+  /**
```

Comment:
Is there a way to override onConnect? I'm not sure about java inheritance. This way lastKnowMqttStatusCode would be updated there. I guess that runOnMqttLoop is called soon after.

Reply:
There currently is already an implementation of `onConnect` in the connector. Additionally, there would be no need to add an update there, though, as the `lastKnownMqttStatusCode` is updated in the general `onStatus` method instead to ensure it captures all status updates, not just connecting.